src/Mvc/Mvc.Analyzers/src/Startup/StartupAnalyzer.Diagnostics.cs

```diff
@@ -16,5 +16,15 @@ public partial class StartupAnalzyer : DiagnosticAnalyzer
             DiagnosticSeverity.Warning, 
             isEnabledByDefault: true,
             helpLinkUri: "https://aka.ms/YJggeFn");
+
+        internal readonly static DiagnosticDescriptor BuildServiceProviderShouldNotCalledInConfigureServicesMethod = new DiagnosticDescriptor(
+            "MVC1007",
```

Comment:
@Tratcher - raised an interesting point. Right now these build from inside the MVC solution - I'm planning to move these to a more general place, and I'll handle the update of this ID when I do it.